tests/components/template/test_alarm_control_panel.py

```diff
@@ -0,0 +1,560 @@
+                        "arm_away": {
+                            "service": "alarm_control_panel.alarm_arm_away",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_home": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_night": {
+                            "service": "alarm_control_panel.alarm_arm_night",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "disarm": {
+                            "service": "alarm_control_panel.alarm_disarm",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    state = hass.states.get("alarm_control_panel.test_template_panel")
+    assert state is not None
+
+    assert state.attributes.get("friendly_name") == "Template Alarm Panel"
+
+
+async def test_arm_home_action(hass):
+    """Test arm home action."""
+    await setup.async_setup_component(
+        hass,
+        "alarm_control_panel",
+        {
+            "alarm_control_panel": {
+                "platform": "template",
+                "panels": {
+                    "test_template_panel": {
+                        "value_template": "{{ states('alarm_control_panel.test') }}",
+                        "arm_away": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_home": {"service": "test.automation"},
+                        "arm_night": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "disarm": {
+                            "service": "alarm_control_panel.alarm_disarm",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    dummy_service = MagicMock()
+    hass.services.async_register("test", "automation", dummy_service)
```

Comment:
We have this helper:
https://github.com/home-assistant/home-assistant/blob/120a96ff5994676e5e478407cf695578d5e004d8/tests/common.py#L232-L243

Reply:
I'm a team player - duly adopted :)